producer/src/signature.rs

```diff
@@ -1,10 +1,9 @@
-use anyhow::Result;
 use identity_iota::storage::JwkStorage;
 
 use crate::SecretManager;
 
 impl SecretManager {
-    pub async fn sign(&self, data: &[u8]) -> Result<Vec<u8>> {
+    pub async fn sign(&self, data: &[u8]) -> Result<Vec<u8>, std::io::Error> {
```

Comment:
Shouldn't this be `Result<Vec<u8>, ProducerError>`?

With an added `ProducerError` variant so that the `unwraps` in this function can be replaced with `?`, e.g.:
```rust
    #[error("Identity error: `{0}`")]
    IdentityCoreSingleStructError(
        #[from] identity_iota::core::SingleStructError<identity_iota::storage::KeyStorageErrorKind>,
    ),
```